oauthenticator/openshift.py

```diff
@@ -7,18 +7,23 @@
 
 import json
 import os
-import requests
+import warnings
 
-from tornado.auth import OAuth2Mixin
+import requests
+from jupyterhub.auth import LocalAuthenticator
 from tornado import web
-
+from tornado.auth import OAuth2Mixin
+from tornado.httpclient import AsyncHTTPClient, HTTPClient, HTTPRequest
 from tornado.httputil import url_concat
-from tornado.httpclient import HTTPRequest, AsyncHTTPClient, HTTPClient
-from traitlets import Bool, Unicode, Set, default
 
-from jupyterhub.auth import LocalAuthenticator
+from oauthenticator.oauth2 import OAuthenticator
+from traitlets import Bool, Set, Unicode, default
 
-from .oauth2 import OAuthenticator
+try:
```

Comment:
@vpavlin in my testing, at least on our UBI based images using the `pycurl` backend fixes any issues we have been having with self-signed certs. Honestly this makes sense considering I couldn't reproduce the issue with standard curl. From what I can tel this isn't pulled in as a dependency automatically by tornado so we'll have to add a step to our build to compile it w/ `nss` support.

Reply:
Yeah, that makes sense. The change looks good to me and I'll make sure the right packages are installed in our images to support pycurl